docs/features/deploy.md

```diff
@@ -0,0 +1,117 @@
+# AlgoKit Deploy Feature Documentation
+
+Deploy your smart contracts effortlessly to various networks with the AlgoKit Deploy feature. This feature is essential for automation in CI/CD pipelines and for seamless deployment to mainnet or testnet networks.
+
+## Usage
+
+```sh
+$ algokit deploy [OPTIONS] NETWORK
+```
+
+This command deploys smart contracts from an AlgoKit compliant repository to the specified network.
+
+### Options
+
+- `--custom-deploy-command TEXT`: Specifies a custom deploy command. If this option is not provided, the deploy command will be loaded from the `.algokit.toml` file.
+- `--ci`: Skips the interactive prompt for mnemonics. When using this option, mnemonics must be set as environment variables.
+- `--prod`: Skips the warning prompt for deployments to a mainnet.
+- `-h, --help`: Show this message and exit.
+
+## Functionality
+
+AlgoKit requires certain configuration files at the root of your project to successfully perform deployment. Below is the directory layout expected by AlgoKit:
+
+```md
+.
+├── ... (your project files and directories)
+├── .algokit.toml # Configuration file for AlgoKit
+└── .env.[{mainnet|testnet|localnet|betanet|custom}] # (OPTIONAL) Environment variables specific to deployments to a network
+```
+
+### AlgoKit Configuration File
+
+AlgoKit allows you to define your deployment logic by adding an `.algokit.toml`, a file present at the root of your project instantiated with `algokit init`. This file should include deployment commands for the network(s) you wish to target. The syntax is as follows:
+
+```toml
+[deploy.{network_name}]
+command = "deployment-command-for-{network_name}"
+```
+
+#### Example
+
+For instance, if you're aiming to deploy to a testnet, your `.algokit.toml` file should contain:
+
+```toml
+[deploy.testnet]
+command = "poetry run python -m smart_contracts deploy"
+```
+
+### Environment Files
+
+For each network you're deploying to, you need to have a corresponding `.env.[network_name]` file. This file should contain environment variables specific to that network.
```

Comment:
This says you need the file, it’s optional right?

Reply:
@robdmoore see the `note` field right below it explicitly saying its optional